AppSyncRealTimeClient/Support/TaskQueue.swift

```diff
@@ -0,0 +1,30 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#if swift(>=5.5.2)
+
+import Foundation
+
+@available(iOS 13.0, *)
+actor TaskQueue<Success> {
+    private var previousTask: Task<Success, Error>?
+
+    func sync(block: @Sendable @escaping () async throws -> Success) {
+        previousTask = Task { [previousTask] in
+            _ = await previousTask?.result
+            return try await block()
```

Comment:
I _think_ this will swallow the thrown error. Could that be problematic here?

Reply:
Yes, it would, good catch!  This would swallow the error at the block level, so any throwing calls within the block could still be caught and handled.  However, this would not be enforced.  Although all errors were currently caught and handled, this could easily have caused problems in the future.  I just pushed an update to force error handling within the blocks so that any unhandled errors now cause a compilation error.